pkg/cli/cmd/run.go

```diff
@@ -12,13 +14,9 @@ type RunCmd struct {
 }
 
 func (cmd *RunCmd) Run(c *cc.CommonCtx) error {
-	if running, err := dockerx.IsRunning(cmd.ContainerName); running || err != nil {
-		if !running {
-			return ErrNotRunning
-		}
-		if err != nil {
-			return err
-		}
+	err := checkRunning(c, cmd.ContainerName)
+	if err != nil && !errors.Is(err, ErrNotRunning) {
```

Comment:
These conditions are pretty confusing and we have them in multiple places.
I think it would clean things up if we wrapped this logic in a `checkRunStatus` function that would take a `expectedStatus` and return the appropriate error.

It might also make sense to define that function on `cc.CommonCtx`:

```go
type runStatus int
const (
    statusNotRunning runStatus = iota
    statusRunning
)

func (cc *CommonCtx) checkRunStatus(containerName string, expected runStatus) error {
...
}
```